scripts/build-alsa-tools.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2025 Intel Corporation. All rights reserved.
+
+# fail immediately on any errors
+set -e
+
+# Array of ALSA Git repository URLs.  Add or remove repositories as needed.
+declare -a REPOS=(
+    "https://github.com/thesofproject/alsa-lib.git"
+    "https://github.com/thesofproject/alsa-utils.git"
+    # Add more repositories here...
+)
+
+# Commit ID to check for (optional). If specified, the script will update
+# the repository if this commit ID is not found.  Leave empty to skip.
+declare -a COMMIT_ID=(
+    "df8f1cc1ec9d9ee15be5e2c23ad25b9389fd8766"
+    "09550cd393b1a7d307ee6f26637b1ed7bd275e38"
+)
+
+# Directory where repositories will be cloned/updated.
+if [[ -z "$SOF_WORKSPACE" ]]; then
+  # Environment variable is empty or unset so use default
+  BASE_DIR="$HOME/work/sof"
```

Comment:
I think this should be

```
BASE_DIR="$HOME/work"
```

For the default case, the BASE_DIR should point to parent directory of sof as this is the directory assumed in tools/topology/CMakeLists.txt rules:

```
set(SOF_ALSA_TOOLS_DIR "${SOF_ROOT_SOURCE_DIRECTORY}/../tools/bin") 
```

So BASE_DIR should match  "${SOF_ROOT_SOURCE_DIRECTORY}".

This is causing failuress e.g. when we try to use "build-tools.sh -A" in docker in github (and might be a reason why @lgirdwood you "open-coded" the build recipe for github testbench workflow in this PR).

Reply:
This is correct. We are cloning into the ```~/work/sof``` workspace directory with contains the ```~/work/sof/sof``` FW directory. e.g. my SOF workspace directory looks like 
```
~/work/sof$ ls
alsa-lib      build-sof-staging  LICENSE.workspace  ruy                 sof-docs      zephyr
alsa-utils    flatbuffers        modules            sof                 sof-test
build-mtl     gemmlowp           nnlib-hifi4        sof-bin             tflite-micro
build-rimage  keys               README.md          sof.code-workspace  tools
```
The workspace should include all ingredients used to build FW, topologies, tooling i.e. everything a developer needs to create a new module and deploy it.  I'll fix the other makefiles.

Reply:
@lgirdwood Right so "$ZEPHYR_WORKSPACE=work/sof" is terms of 
https://thesofproject.github.io/latest/getting_started/build-guide/build-with-zephyr.html

Ok I didn't realize that, I never use "sof" as the workspace directory name, but indeed then the problem is elsewhere setting the workspace path.